tests/utils/host_blas.cpp

```diff
@@ -63,20 +63,20 @@ void mxpy(void *x, void *y, int len, QudaPrecision precision)
 }
 
 // returns the square of the L2 norm of the vector
-template <typename real_t> inline double norm2(real_t *v, int len)
+template <typename real_t> inline double norm2(real_t *v, int len, bool global)
 {
   double sum = 0.0;
   for (int i = 0; i < len; i++) sum += v[i] * v[i];
-  quda::comm_allreduce_sum(sum);
+  if (global) quda::comm_allreduce_sum(sum);
```

Comment:
I need to think about it a bit, but I think that for the context this is used in (3-d laplace eigensolvers) you don't want to outright disable the full reduction, you only want to reduce in the `x/y/z` plane?

Reply:
The disablement of the global reduction was intentional here, as the point was to defer the global reduction until later and do it once in a padded structure to allow for a single 4-d all reduce to get the correct answer.  Looks like I didn't actually do this properly so I'll fix this.

Reply:
38b1610d52b2503ce467ed1a706e394eda8d8fa1